wolfcrypt/src/dsa.c

```diff
@@ -998,12 +998,12 @@ int wc_DsaVerify(const byte* digest, const byte* sig, DsaKey* key, int* answer)
     int    qSz;
 
     do {
-        if (digest == NULL || sig == NULL || key == NULL || answer == NULL) {
```

Comment:
All we need to do to fix the problem is move this line above the `do {}` (keeping it otherwise unchanged), and have it `return BAD_FUNC_ARG` instead of `ret = BAD_FUNC_ARG; break;`.  No need for a refactor.

Reply:
I did the tricky fix because I have thought the function does not allow early return.

Ok, I fixed it.

And I add another patch to fix the same issue in ```wc_DsaSign()```.

Reply:
Excellent, re identifying the second occurrence of the bad code pattern.  Thanks for finding and fixing these!